### ISSUE TITLE
Conftest doesn't exit with plugin exit code

### ISSUE THREAD

**Issue by iAlexBLR** [open-policy-agent/conftest#741]
### Issue:
I've created a conftest plugin for some custom behaviour, but I noticed that conftest when uses installed plugin always returns 0 exit code even if plugin itself returns another one. 

### Reproduce steps: 

**script.sh**
```
#!/usr/bin/env bash
set -e
exit 1
```

**plugin.yaml**
```
name: script
version: 0.1.0
usage: conftest script
command: ./script.sh
```

### Testing:
```
% ./script.sh
% echo $?
1
%
% conftest plugin install .
% conftest script
% echo $?
0
```

**Comment by jalseth** [comment]
Thanks for raising this. If you have the time, I'd welcome a PR to fix.

**Comment by iAlexBLR** [comment]
@jalseth  unfortunately I'm not that good at programming on Go

**Comment by jpreese** [comment]
@iAlexBLR this is sort of by design, see: https://www.conftest.dev/plugins/ (at the bottom)

It has been a _long_ time since I've dealt with plugins in Conftest, but do recall plugins needing to specify other error codes. I can see how this can be confusing though, and I don't recall the technical reason for it unfortunately.

**Comment by jalseth** [comment]
I disagree with the decision to swallow error codes 1, 2 and treat them as non-errors. This isn't what users would expect, and many/most programs default to an exit code of 1 for all errors. This also includes the kubectl plugin provided as an example: https://github.com/open-policy-agent/conftest/blob/c149d816bb161496cdb2402a720fa5e291236690/contrib/plugins/kubectl/kubectl-conftest.sh#L11
